pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -913,9 +935,21 @@ func buildLocalityLbEndpointsFromShards(
 				}
 				localityEpMap[ep.Locality] = locLbEps
 			}
-			if ep.EnvoyEndpoint == nil {
-				ep.EnvoyEndpoint = buildEnvoyLbEndpoint(ep.UID, ep.Family, ep.Address, ep.EndpointPort, ep.Network,
-					ep.LbWeight, ep.TLSMode, push)
+			// labels.Collection seems really not needed...
+			ns := ep.Attributes.Namespace
+			canMtls := push.EndpointAcceptMtls(ns, labels.Collection{ep.Labels}, ep.EndpointPort)
+			// Update the label to disabled when the endpoint is not serving mTLS per policy.
+			if !canMtls {
+				ep.TLSMode = model.DisabledTLSModeLabel
```

Comment:
this is not right as it prevents end users from overriding this in service entries

Reply:
`canMtls` is the result of evaluating beta mTLS policy. even if the service entry has the `tlsMode: istio`, if canMtls is false, the server endpoint LDS is configured with plain text, we can send mTLS.

Do i miss anything?